utilities/test_suite/rpp_test_suite_common.h

```diff
@@ -0,0 +1,287 @@
+
+        for (int i = 0; i < descPtr->h; i++)
+        {
+            for (int j = 0; j < descPtr->w; j++)
+            {
+                *outputTemp = *outputCopyTempR;
+                outputTemp++;
+                outputCopyTempR++;
+                *outputTemp = *outputCopyTempG;
+                outputTemp++;
+                outputCopyTempG++;
+                *outputTemp = *outputCopyTempB;
+                outputTemp++;
+                outputCopyTempB++;
+            }
+        }
+
+        outputCopyTemp += descPtr->strides.nStride;
+    }
+
+    free(outputCopy);
+}
+
+void convert_pkd3_to_pln3(Rpp8u *input, RpptDescPtr srcDescPtr)
+{
+    unsigned long long bufferSize = ((unsigned long long)srcDescPtr->h * (unsigned long long)srcDescPtr->w * (unsigned long long)srcDescPtr->c * (unsigned long long)srcDescPtr->n) + (unsigned long long)srcDescPtr->offsetInBytes;
+    Rpp8u *inputCopy = (Rpp8u *)calloc(bufferSize, sizeof(Rpp8u));
+    memcpy(inputCopy, input, bufferSize * sizeof(Rpp8u));
+
+    Rpp8u *inputTemp, *inputCopyTemp;
+    inputTemp = input + srcDescPtr->offsetInBytes;
+
+    omp_set_dynamic(0);
+    #pragma omp parallel for num_threads(srcDescPtr->n)
+    for (int count = 0; count < srcDescPtr->n; count++)
+    {
+        Rpp8u *inputTempR, *inputTempG, *inputTempB;
+        inputTempR = inputTemp + count * srcDescPtr->strides.nStride;
+        inputTempG = inputTempR + srcDescPtr->strides.cStride;
+        inputTempB = inputTempG + srcDescPtr->strides.cStride;
+        Rpp8u *inputCopyTemp = inputCopy + srcDescPtr->offsetInBytes + count * srcDescPtr->strides.nStride;
+
+        for (int i = 0; i < srcDescPtr->h; i++)
+        {
+            for (int j = 0; j < srcDescPtr->w; j++)
+            {
+                *inputTempR = *inputCopyTemp;
+                inputCopyTemp++;
+                inputTempR++;
+                *inputTempG = *inputCopyTemp;
+                inputCopyTemp++;
+                inputTempG++;
+                *inputTempB = *inputCopyTemp;
+                inputCopyTemp++;
+                inputTempB++;
+            }
+        }
+    }
+
+    free(inputCopy);
+}
+
+template <typename T>
+void compare_output(T* output, string func, string funcName, RpptDescPtr srcDescPtr)
+{
+    bool isEqual = false;
+    string ref_path = get_current_dir_name();
+    string pattern = "HOST_NEW/build";
+    remove_substring(ref_path, pattern);
+    string ref_file = ref_path + "REFERENCE_OUTPUT/" + funcName + "/"+ func + ".csv";
+    ifstream file(ref_file);
+
+    vector<vector<string>> refOutput;
+    vector<string> row;
+    string line, word;
+    if(file.is_open())
+    {
+        while(getline(file, line))
+        {
+            row.clear();
+            stringstream str(line);
+            while(getline(str, word, ','))
+            row.push_back(word);
+            refOutput.push_back(row);
+        }
+    }
+    else
+        cout<<"Could not open the file\n";
+
+    for(int i = 0; i < refOutput.size(); i++)
```

Comment:
This loop can probably be run more efficiently using Rpp styled addressing. Pls dont compute srcDescPtr->strides.hStride * i again and again in loop. Also, set isEqualto be true initially. If anything doesn't match, just set to false and remove the else condition. Also, please add a tolerance of +/- X%, where X can be taken from user for now.

Reply:
The other metric that's useful is %match of the pixels. In that case, you need to go over the whole image, but the final pass or fail must be based on whether each pixel is within tolerance